packages/client/hmi-client/src/components/model/tera-parameters-metadata.vue

```diff
@@ -1,5 +1,49 @@
 <template>
-	<ul>
+	<template v-if="isStratified">
+		<ul>
+			<li
+				v-for="([baseParameterId, values], index) in collapsedParameters.entries()"
+				:key="baseParameterId"
+			>
+				<template v-if="values.length > 1">
+					<tera-parameter-metadata-entry
+						:parameter="model.metadata?.parameters?.[baseParameterId] ?? { id: baseParameterId }"
```

Comment:
I see quite a few of thees conditional assignment and emit patterns that I have a concern with.

```
<something
  :parameter = "model.foobar.blah[xyz] ?? { id: "abc" }"
  @update-parameter = update(xyz) >
</something>
```

This is a bit difficult to discern exactly what is getting updated, and I don't know why we need to have an alternate path `?? {id "abc"}`

I also see in quite a few places you are doing `parameters.find(...)`, this not very efficient. And it may be the case that we should consider preprocessing the data into a more template-friendly data structure rather than using it as-is.

Reply:
For that case I am updating the base parameter. The base parameter doesn't exist in the model so I have to create it in the metadata. So first it is `{ id: baseParameterId }` and once `@update-parameter` is ran it'll be referenced in the metadata as `model.metadata?.parameters?.[baseParameterId]`

Reply:
I thought specifying the update type would be a pain because I'd be passing too many emits

Reply:
I think I need a refresher on why are we creating phantom variables that do not exist.....this might be a chicken-and-egg problem, let's maybe sync up.